### PR TITLE
Modular repo

### PR DESCRIPTION
**Pull request by pavelpicka**
Add modular repository with modulemd, modulemd defaults, and modulemd
obsoletes with real-world packages and metadata.

[noissue]

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,6 +91,11 @@ help:
 	@echo "        Create an RPM repository with modules"
 	@echo "    fixtures/rpm-with-modules-modified"
 	@echo "        Create an RPM repository with modules but without some dependencies."
+	@echo "    fixtures/rpm-modular"
+	@echo "        Create an RPM repository with modules, module defaults and module obsoletes."
+	@echo "        Packages taken from dnf-ci-fedora-modular repository and combined with "
+	@echo "        few modular metadata from dnf-ci-thirdparty-modular and dnf-ci-obsoletes"
+	@echo "        as those metadata is intended to operate above those packages."
```
Comment (dralley) [inline Makefile:98]: Could you add more detail about what modular metadata was copied?  Or was it all of it

File: Makefile [hunk 2]
```diff
@@ -91,6 +91,11 @@ help:
 	@echo "        Create an RPM repository with modules"
 	@echo "    fixtures/rpm-with-modules-modified"
 	@echo "        Create an RPM repository with modules but without some dependencies."
+	@echo "    fixtures/rpm-modular"
+	@echo "        Create an RPM repository with modules, module defaults and module obsoletes."
+	@echo "        All packages taken from dnf-ci-fedora-modular repository and combined with "
+	@echo "        all modular metadata from dnf-ci-thirdparty-modular and dnf-ci-obsoletes"
+	@echo "        as those metadata is intended to operate above those packages."
```
Comment (dralley) [inline Makefile:98]: Oh so the modular metadata is meant to go with the packages from the other repo, they're not for a separate group of packages?

Comment (dralley) [reply Makefile:98]: Not necessarily here, but I would like to have the URLs documented as well.

File: Makefile [hunk 3]
```diff
@@ -233,6 +238,7 @@ all-fedora: \
 	fixtures/rpm-with-non-utf-8 \
 	fixtures/rpm-alt-layout \
 	fixtures/rpm-with-modules \
+	fixtures/rpm-modular \
 	fixtures/rpm-with-modules-modified \
 	fixtures/rpm-modules-static-context \
```
Comment (dralley) [inline Makefile:243]: Let's plan to get rid of some of these older fixture repos once the tests have been updated.

File: Makefile (hunk 2 above)
Comment (pavelpicka) [reply Makefile:98]: Yes, those modular data are meant directly for the packages taken/used.

Comment (pavelpicka) [reply Makefile]: I moved the whole message to the README file with a link to the original repository.

File: Makefile (hunk 3 above)
Comment (pavelpicka) [reply Makefile:240]: I concur.

File: Makefile [hunk 4]
```diff
@@ -91,6 +91,8 @@ help:
 	@echo "        Create an RPM repository with modules"
 	@echo "    fixtures/rpm-with-modules-modified"
 	@echo "        Create an RPM repository with modules but without some dependencies."
+	@echo "    fixtures/rpm-modular"
+	@echo "        Create an RPM repository with modules, module defaults and module obsoletes."
```
Comment (dralley) [inline Makefile:95]: Perhaps we could talk with them about having `static-context` covered as well?  I don't see any modules using it in here.

**Review by dralley — APPROVED** [review]
Approved: further improvements (e.g. static-context) can be added later

**Comment by dralley** [comment]
You might need to ask Dennis, Tanya, Ina, etc. to merge.  I don't have the ability.

File: Makefile (hunk 4 above)
Comment (pavelpicka) [reply Makefile:95]: I adjusted [gen-patched-fixtures.sh](https://github.com/pulp/pulp-fixtures/pull/230/files#diff-bb6476db87bc921265dfdb517b7ee0e36009a5b36c98086e68d7dfc2f619825f) and patch for ``static-context`` as well that we can test this field with new fixtures.

**Review by fao89 — APPROVED** [review]
LGTM

**Comment by fao89** [comment]
> You might need to ask Dennis, Tanya, Ina, etc. to merge. I don't have the ability.

you have it now

**Comment by pavelpicka** [comment]
Yes, there is already an open PR to fix it.

**Pull request merged**